main.tf

```diff
@@ -46,7 +46,7 @@ resource "aws_security_group_rule" "bastion_all_egress" {
 }
 
 data "template_file" "user_data" {
-  template = "${file("${path.module}/${var.user_data_file}")}"
+  template = "${var.user_data_file != "" ? var.user_data_file : file("${path.module}/user_data.sh")}"
```

Comment:
Let's keep this code backward compatible because it is possible to do so - keep `user_data_file` variable. Check if content of `user_data_template` ( a new variable which you should define) is not empty then use it, otherwise use `file(...)`, as before.

Also make respective changes in README.

Reply:
@antonbabenko I see the solution in a different way: if filename is empty then use default user_data.sh script. The reason was to give way for customization of the whole config file. What is should a variable `user_data_template` stand for?

Reply:
`user_data_template` - contains template itself. `user_data_file` - name of the file to get content from.

If possible, always make backward compatible changes. Here it is possible.